### ISSUE TITLE
Umpredictable unstable (buggy?) SVD evaluation

### ISSUE THREAD

**Issue by Nekrolm** [dimforge/nalgebra#983]
Hi! I have faced with strange behavior of `Matrix::svd` method:

Largest singular value somehow significantly depends on `compute_u`/`compute_v` params. And transposition can hide/expose this effect.

There is an example (nalgebra = 0.28, also reproducible for 0.29)
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d96e388cd6e8aa7f437451d44960b9fe

```
    let svd1 = m.svd(true, true);
    println!("{:?}", svd1.singular_values);
    let svd2 = m.svd(false, false);
    println!("{:?}", svd2.singular_values);
    let svd3 = m.svd(false, true);
    println!("{:?}", svd3.singular_values);
    let svd4 = m.svd(true, false);
    println!("{:?}", svd4.singular_values);
```
prints different values for same matrix:
```
Matrix { data: [[0.0000000000000034441735216135435, 46.56197615440326, 22.38119784648299]] }
Matrix { data: [[0.000000000000002338832028974828, 31.618802153516995, 22.38119784648299]] }
Matrix { data: [[0.0000000000000020737891983235553, 28.035673172576868, 22.38119784648299]] }
Matrix { data: [[0.0000000000000034441735216135435, 46.56197615440326, 22.38119784648299]] }
```

It does not look like numerical error.

**Comment by Nekrolm** [comment]
result from [`numpy`](https://colab.research.google.com/drive/1TjX1NQcSfGCimmOmlNmgliVlTOe4Cl8b?usp=sharing)

```
A = np.array([[10.74785316637712, -5.994983325167452, -6.064492921857296], 
              [-4.149751381521569, 20.654504205822462, -4.470436210703133], 
              [-22.772715014220207, -1.4554372570788008, 18.108113992170573]])
U, S, Vt = np.linalg.svd(A)
print(S)
```
```
[3.16188022e+01 2.23811978e+01 3.31763547e-15]
```

**Comment by cfunky** [comment]
The particular case of 3x3 statically allocated matrices seems to have been fixed, probably with release 0.30.0. For dynamically allocated matrices the problem still persists.

**Comment by Andlon** [comment]
Fixed by #1089.
